app/components/paperShow/components/betterSearch.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from "react";
+import * as classNames from "classnames";
+import { debounce } from "lodash";
+import { History } from "history";
+import { connect, Dispatch } from "react-redux";
+import { withStyles } from "../../../helpers/withStylesHelper";
+import { Fos } from "../../../model/fos";
+import Icon from "../../../icons";
+import InputWithSuggestionList from "../../common/InputWithSuggestionList";
+import papersQueryFormatter from "../../../helpers/papersQueryFormatter";
+import ActionTicketManager from "../../../helpers/actionTicketManager";
+import { ACTION_TYPES } from "../../../actions/actionTypes";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import { CompletionKeyword } from "../../../api/completion";
+import { getKeywordCompletion } from "../../layouts/actions";
+import EnvChecker from "../../../helpers/envChecker";
+const s = require("./betterSearch.scss");
+
+interface BetterSearchProps {
+  FOSList: Fos[];
+  suggestionKeywords: CompletionKeyword[];
+  isAnimated: boolean;
+  dispatch: Dispatch<any>;
+}
+
+function handleSubmitQuery(query: string, dispatch: Dispatch<any>, history: History) {
+  if (query.length < 2) {
+    return dispatch({
+      type: ACTION_TYPES.GLOBAL_ALERT_NOTIFICATION,
```

Comment:
Why did not you use the alertToast helper?

Reply:
I'll remove that function later.
`alertToast` will be deprecated

Reply:
OK!